utils.js

```diff
@@ -4321,42 +4321,118 @@ export const parseCsProjData = async function (csProjData) {
   }
-  for (const alib of Object.keys(assetData.libraries)) {
-    // Skip os runtime packages
-    if (alib.startsWith("runtime")) {
-      continue;
+
+  csProjData = JSON.parse(csProjData);
+  rootPkg =
+    {
+      group: "",
+      name: csProjData.project.restore.projectName,
+      version: csProjData.project.version || "latest",
+      type: "application",
+      "bom-ref": decodeURIComponent(
+        new PackageURL(
+          "nuget",
+          "",
+          csProjData.project.restore.projectName,
+          csProjData.project.version || "latest",
+          null,
+          null
+        ).toString()
+      )
     }
-    const tmpA = alib.split("/");
-    const libData = assetData.libraries[alib];
-    if (tmpA.length > 1) {
-      pkg = {
-        group: "",
-        name: tmpA[0],
-        version: tmpA[tmpA.length - 1]
-      };
-      if (libData.sha256) {
-        pkg._integrity = "sha256-" + libData.sha256;
-      }
-      if (libData.sha512) {
-        pkg._integrity = "sha512-" + libData.sha512;
+
+  const purlString = decodeURIComponent(rootPkg["bom-ref"].toString());
+  pkgList.push(rootPkg);
+  let rootPkgDeps = new Set();
+  const depList = new Set();
+  if (
+    csProjData.libraries &&
+    csProjData.targets
+  ) {
+    const lib = csProjData.libraries;
+    for (const framework in csProjData.targets) {
+      for (const rootDep of Object.keys(csProjData.targets[framework])) {
+        const [group, name] = rootDep.split("/");
+        const dpurl = decodeURIComponent(
+          new PackageURL(
+            "nuget",
+            "",
+            group,
+            name,
+            null,
+            null
+          ).toString()
+        );
+        let pkg = {
+            name: group,
+            version: name,
+            description: "",
+            type: csProjData.targets[framework][rootDep].type,
+            "bom-ref": dpurl,
+          }
+        if (lib[rootDep]) {
+          pkg["type"] = "library"
+          if (lib[rootDep].sha512){
+            pkg["_integrity"] = "sha512-" + lib[rootDep].sha512;
+          }
+          else if (lib[rootDep].sha256){
+            pkg["_integrity"] = "sha256-" + lib[rootDep].sha256;
+          }
+        }
+        pkgList.push(pkg);
+        rootPkgDeps.add(dpurl);
+
+        const dependencies = csProjData.targets[framework][rootDep].dependencies;
+        if (dependencies) {
+          Object.keys(dependencies).forEach((p) => {
+            const ipurl = decodeURIComponent(
+              new PackageURL(
+                "nuget",
+                "",
+                p,
+                dependencies[p],
+                null,
+                null
+              ).toString()
+            );
+            depList.add(ipurl);
+            pkgList.push(
+              {
+                group: "",
+                name: p,
+                version: dependencies[p],
+                description: "",
+                "bom-ref": ipurl
```

Comment:
bom-ref must use decodeURIComponent so `decodeURIComponent(ipurl)`.

Reply:
@prabhu ipurl is a const created with decodeURIComponent already.